pkg/kubelet/images/puller_test.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package images
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	"k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+// FakeImageService is a fake implementation of the ImageService interface that returns empty values.
+type FakeImageService struct {
+	pullImageFn func(ctx context.Context, image container.ImageSpec, secrets []v1.Secret, podSandboxConfig *runtimeapi.PodSandboxConfig) (string, error)
+}
```

Comment:
Can we use/modify [staging/src/k8s.io/cri-api/pkg/apis/testing/fake_image_service.go](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cri-api/pkg/apis/testing/fake_image_service.go) instead of defining a new struct?

Reply:
this type is very specific for this test to allow to inject a function to sync the parallel executions, trying to consolidate will add a cross dependency and complicate the code , it looks better to have duplicate code in this case